api/aws/tfvars.go

```diff
@@ -26,7 +28,10 @@ func (e *Cluster) AddMachine(
 	machine *api.Machine,
 ) (string, error) {
 	if name == "" {
-		name = uuid.New().String()
+		// TODO Find the root cause for this issue
+		name = strings.Replace(uuid.New().String(), "-", "", -1)
+		// TODO Create a more dynamic workaround for the too long names
+		name = name[:10]
```

Comment:
There is a risk here (albeit minimal) with name collisions. @Xartos if your fix isn't too far away I'd rather just wait for that otherwise we can use this as a temporary workaround if it's causing issues.